subsys/net/ip/net_core.c

```diff
@@ -61,7 +62,7 @@ static inline enum net_verdict process_data(struct net_pkt *pkt,
 	int ret;
 	bool locally_routed = false;
 
-	ret = net_packet_socket_input(pkt);
+	ret = net_packet_socket_input(pkt, ETH_P_ALL);
 	if (ret != NET_CONTINUE) {
 		return ret;
 	}
```

Comment:
If we do not have Ethernet enabled, I just wonder why would we actually need to do this in the first place.
@lmajewski would the changes here still work with DSA, can you review the code?

Reply:
The code looks generally OK, however I'm concern that `ETH_P_ALL` and `IPPROTO_RAW` use some common logic (like `raw_pkt_continue` flag) in the `connection.c` and hence I will need to test if those patches don't break DSA.

Another issue is the level of `connection.c` file convolution.